src/vendors/common_func.js

```diff
@@ -220,6 +220,69 @@
     });
   };
 
+  window.promoCarousel = function() {
```

Comment:
Why do you defined this here and in the common_func as well? Use common_func for general functions such as window object. Vue component should contain only vue elements

Reply:
This approach was used in other vue components too. I needed as general rule to have access to customize only some specific carousels.